src/os/os.c

```diff
@@ -313,17 +313,32 @@ static void os_thread_start(void *arg)
     _impure_ptr = priv->reent;
     (*priv->entry)(priv->arg);
 
-    TaskList *current = &taskList;
-    TaskList *last = &taskList;
     vTaskSuspendAll();
-    while (current->task != xTaskGetCurrentTaskHandle())
+    TaskList *current = taskList;
```

Comment:
you can write this whole code simpler by using an iteration variable like this:
TaskList** prev_ptr = &taskList;
 Then you don't need special case for single, first and last entries.